### PR TITLE
Improve deed die mechanic - fixes #215

### LINKED ISSUE #215: Add a simpler means of using Deed dice
Some folks find the Deed dice system hard to follow. Maybe they forget to click it because it is separate from attacks. Maybe an option to prompt before an attack roll if you have selected Mighty Deeds on your sheet config?

### PR DESCRIPTION
**Pull request by radoslawg**
Maybe fixes #215

This is my attempt at improving deed die mechanic. I understand why Deed Die in DCC Ruleset is implemented as is, and I understand developers do not want to "take stance" (quoted from Discord). What I did is, I gave options to groups that want to roll Deed Die with every attack. 

Reason behind this is that if player forgets to roll die with each round then value from previous round is added to results, it takes group out of action to recalculate manually rolls (granted, not complicated math but it takes a bit of time). Also, it gives option to groups that understands that core book rules mean to roll deed die with every attack.

This pull request also integrates deed die with DCC roll result cards. I tried to follow what I see is implemented already but javascript is not language I use every day.

### CODE REVIEW

File: module/actor.js [hunk 1]
```diff
@@ -112,6 +112,7 @@ class DCCActor extends Actor {
       capLevel: false,
       maxLevel: 0,
       rollAttackBonus: false,
+      rollABWithAttack: false,
```
Comment (TylerKostuch) [inline module/actor.js:115]: Bit of a nit pick, but the variable name uses the abbreviation "AB" whereas the line before spells it out. AB isn't a super common abbreviation, so consider renaming the variable `rollAttackWithAttackBonus`?

File: module/actor.js [hunk 2]
```diff
@@ -527,7 +545,11 @@ class DCCActor extends Actor {
         fumbleResult = await this.rollFumble(options)
       }
 
-      const emote = options.backstab ? 'DCC.BackstabEmote' : 'DCC.AttackRollEmote'
+      let emote = options.backstab ? 'DCC.BackstabEmote' : 'DCC.AttackRollEmote'
+      if (config.rollAttackBonus && config.rollABWithAttack) {
+        emote = emote.concat("Deed");
```
Comment (TylerKostuch) [inline module/actor.js:550]:
```suggestion
        emote += 'Deed';
```

Not sure if I like the concatenation here rather than just having explicit branching logic, but it works.

File: lang/en.json [hunk 3]
```diff
@@ -64,9 +64,11 @@
   "DCC.AsCoins": "As Coins",
   "DCC.Attack": "Attack",
   "DCC.AttackBonus": "Attack Bonus",
+  "DCC.AttackBonusRollInvalidFormulaInline": "<b>Invalid Attack Bonus formula '{formula}'</b>",
   "DCC.AttackPl": "Attacks",
   "DCC.AttackRoll": "Attack Roll with {weapon}",
   "DCC.AttackRollEmote": "Attacks with their {weaponName} and hits AC {rollHTML} for {damageRollHTML} points of damage!{crit}{fumble}",
+  "DCC.AttackRollEmoteDeed": "Attacks with their {weaponName} and hits AC {rollHTML} with deed die of {attackBonus} for {damageRollHTML} points of damage!{crit}{fumble}",
```
Comment (TylerKostuch) [inline lang/en.json:71]:
```suggestion
  "DCC.AttackRollEmoteDeed": "Attacks with their {weaponName} and hits AC {rollHTML} with a deed roll of {attackBonus} for {damageRollHTML} points of damage!{crit}{fumble}",
```

Super minor, but I think "deed die" generally refers to the actual die (i.e. d3, d5, etc) rather than the value of the roll.

File: lang/en.json [hunk 4]
```diff
@@ -75,6 +77,7 @@
   "DCC.Backstab": "Backstab",
   "DCC.BackstabRoll": "Backstab Roll with {weapon}",
   "DCC.BackstabEmote": "Backstabs with their {weaponName} and hits AC {rollHTML} for {damageRollHTML} points of damage!{crit}{fumble}",
+  "DCC.BackstabEmoteDeed": "Backstabs with their {weaponName} and hits AC {rollHTML} with deed die of {attackBonus}, for {damageRollHTML} points of damage!{crit}{fumble}",
```
Comment (TylerKostuch) [inline lang/en.json:80]:
```suggestion
  "DCC.BackstabEmoteDeed": "Backstabs with their {weaponName} and hits AC {rollHTML} with a deed roll of {attackBonus}, for {damageRollHTML} points of damage!{crit}{fumble}",
```

See above comment

**Review by TylerKostuch — COMMENTED** [review]
Here's my two cents

Also need to look at that check that failed and figure out what's going on there.

I haven't ran the code yet but I'll be merging it into my fork. I've been using the macro that I posted in the issue but it slows things down when the warrior wants to pick up a new weapon and I have to modifier a macro for them.

**Comment by mooped** [comment]
Thanks for submitting this.

I've merged it into a local branch for some additional changes building on it (and also to deal with conflicts with recent changes on the dev branch).
